### PR TITLE
[tf.keras] Add missing cast in keras/layers/normalization.py

### PR DESCRIPTION
**Pull request by georgepaw**
`scale` needs to be casted into the right data type

### CODE REVIEW

**Comment by georgepaw** [comment]
This pull request fixes it as well https://github.com/tensorflow/tensorflow/pull/20809
